### PR TITLE
docs: showcase updated use cases in example workflows

### PR DESCRIPTION
**Pull request by zimeg**
### Summary

This PR updates the example workflows included with this action, and focuses these on common cases 📚

Closes: #158, #270, #349, #361

### Preview

Changes in progress: https://github.com/slackapi/slack-github-action/tree/zimeg-docs-examples/example-workflows

### Reviewers

All of these workflows have been tested in **real** GitHub Actions workflows! Including the release `announcements`!

Please feel free to review these, but I'd like to request specific reviews for the `README.md` documents included with techniques 🙏 ✨ 

### Requirements

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/slack-github-action/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: example-workflows/Technique_2_Slack_API_Method/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Technique 2: Slack API method
+
+A bot token or user token or [token of some other kind][tokens] must be used to
```
Comment (lukegalbraithrussell) [inline example-workflows/Technique_2_Slack_API_Method/README.md:3]: wait, so any kind of token is fair game?

File: example-workflows/Technique_2_Slack_API_Method/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+call one of [the Slack API methods][methods] with this technique.
+
+## Setup
+
+For details on how to setup this technique in GitHub Actions, read the
```
Comment (lukegalbraithrussell) [inline example-workflows/Technique_2_Slack_API_Method/README.md:8]:
```suggestion
For details on how to set up this technique in GitHub Actions, read the
```

File: example-workflows/Technique_2_Slack_API_Method/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+## Example workflows
+
+1. [**Direct message the author**](#direct-message-the-author): Write to the
+   Slack user with a matching email.
+2. [**Invite a usergroup to channel**](#invite-a-usergroup-to-channel): Create a
+   channel and invite members.
+
+### Direct message the author
+
+Send a direct message to the user that pushed the most recent commits.
```
Comment (lukegalbraithrussell) [inline example-workflows/Technique_2_Slack_API_Method/README.md:20]:
```suggestion
Send a direct message to the user that pushed the most recent commit.
```

I think? Unless it's like "send it to the users who pushed the last 5 committs"

File: example-workflows/Technique_2_Slack_API_Method/README.md (hunk 1 above)
Comment (zimeg) [reply example-workflows/Technique_2_Slack_API_Method/README.md:3]: Yep! What'ever's passed in here is sent with the API method used 🔍

File: example-workflows/Technique_2_Slack_API_Method/README.md (hunk 3 above)
Comment (zimeg) [reply example-workflows/Technique_2_Slack_API_Method/README.md:20]: This is wild, but a `push` might actually contain multiple commits 😳  Super confusing for me!

File: example-workflows/Technique_3_Slack_Incoming_Webhook/README.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+## Example workflows
+
+1. [Post an inline text message](#post-an-inline-text-message)
+2. [Post an inline block message](#post-an-inline-block-message)
+3. [Post blocks found in a file](#post-blocks-found-in-a-file)
+
+### Post an inline text message
+
+Write a single world after a push event is received.
```
Comment (lukegalbraithrussell) [inline example-workflows/Technique_3_Slack_Incoming_Webhook/README.md:24]:
```suggestion
Write a single word after a push event is received.
```

Comment (lukegalbraithrussell) [reply example-workflows/Technique_3_Slack_Incoming_Webhook/README.md:24]: although it isn't just one word? that would be a weird limitation maybe?

File: example-workflows/Technique_3_Slack_Incoming_Webhook/blocks.yml [hunk 5]
```diff
@@ -1,34 +1,36 @@
...
                 text:
                   type: "mrkdwn"
                   text: "Danny Torrence left the following review for your property:"
               - type: "section"
-                block_id: "section567"
-                text: {
+                text:
                   type: "mrkdwn"
-                  text: "<https://example.com|Overlook Hotel> \\n :star: \\n Doors had too many axe holes, guest in room 237 was far too rowdy, whole place felt stuck in the 1920s."
+                  text: "<https://example.com|*Overlook Hotel*> :star: \n Doors had too many axe holes, guest in room 237 was far too rowdy, whole place felt stuck in the 1920s."
```
Comment (lukegalbraithrussell) [inline example-workflows/Technique_3_Slack_Incoming_Webhook/blocks.yml:28]: lol

File: example-workflows/Technique_3_Slack_Incoming_Webhook/blocks.yml [hunk 6]
```diff
@@ -1,34 +1,36 @@
...
                 accessory:
                   type: "image"
                   image_url: "https://is5-ssl.mzstatic.com/image/thumb/Purple3/v4/d3/72/5c/d3725c8f-c642-5d69-1904-aa36e4297885/source/256x256bb.jpg"
                   alt_text: "Haunted hotel image"
               - type: "section"
-                block_id: "section789"
                 fields:
                   - type: "mrkdwn"
-                    text: "*Average Rating*\\n1.0"
+                    text: "*Average Rating*: 1.0"
```
Comment (lukegalbraithrussell) [inline example-workflows/Technique_3_Slack_Incoming_Webhook/blocks.yml:36]: but the viewssssss

File: example-workflows/Technique_2_Slack_API_Method/README.md (hunk 1 above)
Comment (lukegalbraithrussell) [inline example-workflows/Technique_2_Slack_API_Method/README.md:3]:
```suggestion
A [token][tokens], such as a bot or user token, must be used to
```

File: example-workflows/Technique_2_Slack_API_Method/README.md (hunk 3 above)
Comment (lukegalbraithrussell) [reply example-workflows/Technique_2_Slack_API_Method/README.md:20]: oh does "recent" consider all commits in that push as coming in at the same time?

File: example-workflows/Technique_1_Slack_Workflow_Builder/README.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
+# Technique 1: Slack Workflow Builder
+
+> :memo: This technique requires [a Slack paid plan][plans] to use Workflow
+> Builder.
```
Comment (lukegalbraithrussell) [inline example-workflows/Technique_1_Slack_Workflow_Builder/README.md:4]:
```suggestion
> :memo: This technique uses Workflow Builder, which requires [a Slack paid plan][plans].
```

File: example-workflows/Technique_1_Slack_Workflow_Builder/README.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+### Post release announcements
+
+Select a channel to post news about the most recent release to.
+
+This example uses [Slack functions][functions] and inline inputs to do the
+following:
+
+1. Open a form to select a channel.
+2. Send a message to the selected channel.
+3. React with an excited emoji.
```
Comment (lukegalbraithrussell) [inline example-workflows/Technique_1_Slack_Workflow_Builder/README.md:49]:
```suggestion
3. React with an `:excited:` emoji.
```

maybe? idk maybe this is obnoxious

**Review by lukegalbraithrussell — COMMENTED** [review]
left some feedback! readme's look great, nothing blocking

File: example-workflows/Technique_1_Slack_Workflow_Builder/README.md (hunk 8 above)
Comment (zimeg) [reply example-workflows/Technique_1_Slack_Workflow_Builder/README.md:49]: I think it's fun! Though `:excited:` might be misleading. I think we're celebrating with a `:tada:` 🎉

Comment (lukegalbraithrussell) [reply example-workflows/Technique_1_Slack_Workflow_Builder/README.md:49]: oh then maybe be explicit with that?
```suggestion
3. React with a `:tada:` emoji.
```

File: example-workflows/Technique_3_Slack_Incoming_Webhook/README.md (hunk 4 above)
Comment (zimeg) [reply example-workflows/Technique_3_Slack_Incoming_Webhook/README.md:24]: Thanks for calling this out!! I forget if I intended to actually write a single word or not, but that's not what the example is showing anyways 😭

Going to update this to something like "a single line of text"!

File: example-workflows/Technique_2_Slack_API_Method/README.md (hunk 3 above)
Comment (zimeg) [reply example-workflows/Technique_2_Slack_API_Method/README.md:20]: Ya! You get both a list of commits **and** the `head_commit` which is, IMO, the more interesting one for this action 🗣️

Comment (lukegalbraithrussell) [reply example-workflows/Technique_2_Slack_API_Method/README.md:20]: i feel like this explains why my green squares aren't appearing -- i'm not great about pushing every day, even if im committing. rip

File: example-workflows/Technique_2_Slack_API_Method/README.md (hunk 2 above)
Comment (zimeg) [reply example-workflows/Technique_2_Slack_API_Method/README.md:8]: I refuse to believe a rules exists for phrasewords like "setup" and "login". Or the rule is "do the opposite of what seemed right" 😉

File: example-workflows/Technique_3_Slack_Incoming_Webhook/blocks.yml (hunk 6 above)
Comment (zimeg) [reply example-workflows/Technique_3_Slack_Incoming_Webhook/blocks.yml:36]: Not so much to see when you're lost in maze! 🦔 ⭐

**Review by lukegalbraithrussell — APPROVED** [review]
readmes look good! let's explore getting this on the tools site next year

**Comment by zimeg** [comment]
@lukegalbraithrussell Thanks for both a review and great suggestions 🙏 ✨ 

Uploading this to tools.slack.dev sounds so helpful for organizing these pages! I await this with excitement 👾

I reran the included examples and am feeling like we can merge these! I'm open to following up with changes of fixes of course, but in this moment this seems to work well 🚀

**Pull request merged**